### PR TITLE
Adding Session Storage Interface

### PR DESCRIPTION
**Pull request by tomarsachin2271**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by AmanRaj1608 — APPROVED** [review]
I was thinking maybe we should another method as `resetState` which sets the data to null so if there is a mismatch in dapp the dapp wil reset the data and agar merkle with just one root value will be set?

**Comment by tomarsachin2271** [comment]
> 

A method to reset the state can be added, but when will the client call this method? I hope this will be checked by the implementation of session storage interface which will check this matching of merkle root in storage with on-chain data

**Review by livingrockrises — COMMENTED** [review]
reviewed

File: packages/modules/src/interfaces/ISessionStorage.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export type SessionStatus = 'PENDING' | 'ACTIVE' | 'INACTIVE' | 'EXPIRED'
+
+export type SessionLeafNode = {
+  validUntil: number
+  validAfter: number
+  sessionValidationModule: string
+  sessionKeyData: string
+  signerPublicKey: string
+  sessionID: string
```
[the remark this answers is not in the document]
Comment (tomarsachin2271) [reply packages/modules/src/interfaces/ISessionStorage.ts]: Done.

File: packages/modules/src/interfaces/ISessionStorage.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  status: SessionStatus
+}
+
+export type SessionSearchParam = {
```
[the remark this answers is not in the document]
Comment (tomarsachin2271) [reply packages/modules/src/interfaces/ISessionStorage.ts:15]: When user will create an instance of SessionLocalStorage then in that constructor user will pass SCW address and the json data that'll be stored in the storage will be against that scw address.

So if user uses a different SCW address tomorrow, then a new key and data will be created in local storage for that user. You'll see once you see the implementation of this interface.

**Review by livingrockrises — APPROVED** [review]
LGTM

**Pull request merged**